### DISCUSSION TITLE
Artworks are Missing with Sonos on HA

### DISCUSSION THREAD

**Discussion by ALaDoffe** [philippe44/AirConnect#519] [Ideas]
Hi 

I dont' know if it is an issue, a limitation, a misconfiguration from my side or maybe a new feature.

I am using airupnp on RPI3 with my Sonos Speakers and everything work well. THX very much @philippe44 
I am also using Home Assistant to control my Sonos Speakers.

When streaming from IOS with AirUpnp, artwork does not appear in HA or in Sonos app. However arstist album song info are on boths apps.
When streaming directly from Sonos APP, all metadata appear in HA.

Also is it way to control (next, previous,...) airplay source  from HA ?
It is working when streaming with sonos app but not with airupnp.

Here below my conf.

``<common>``
``<enabled>1</enabled>``
``<max_volume>100</max_volume>``
``<http_length>-1</http_length>``
``<upnp_max>1</upnp_max>``
``<codec>aac:320</codec>``
``<metadata>1</metadata>``
``<flush>1</flush>``
``<latency>750:1500:f</latency>``
``<drift>0</drift>``
``</common>``
``<main_log>warn</main_log>``
``<upnp_log>warn</upnp_log>``
``<util_log>warn</util_log>``
``<raop_log>warn</raop_log>``
``<log_limit>64</log_limit>``
``<max_players>32</max_players>``
``<binding>?</binding>``
``<ports>0:0</ports>``
``

**Comment by philippe44** [comment]
Have a look at the README, all the limitations on metadata are explained with some options to improve that a bit.
